R/zchunk_L2062.ag_Fert_irr_mgmt.R

```diff
@@ -30,74 +34,140 @@ module_aglu_L2062.ag_Fert_irr_mgmt_DISABLED <- function(command, ...) {
+      replace_na(list(coefficient = 0)) %>%
+
+      # Calculate fertilizer cost using a fixed value (specified in constants.R in 2007$ per ton of NH3)
+      # and the fertilizer coefficient calculated above. Subtract from original nonLandVariableCost.
+      mutate(FertCost = coefficient * aglu.FERT_COST * gdp_deflator(1975, 2007) * CONV_KG_T / CONV_NH3_N ) %>%
+      mutate(nonLandVariableCost = round(nonLandVariableCost - FertCost, aglu.DIGITS_CALPRICE)) %>%
+      select(-minicam.energy.input, -coefficient, -FertCost) ->
+      L2062.AgCost_ag_irr_mgmt_adj
+
+    # Adjust nonLandVariableCost to separate fertilizer cost (which is accounted for specifically)
+    L2052.AgCost_bio_irr_mgmt %>%
+      # Note: using left_join because there are instances with cost but no fertilizer use
+      left_join(L2062.AgCoef_Fert_bio_irr_mgmt,
+                by=c("region", "AgSupplySector", "AgSupplySubsector", "AgProductionTechnology", "year")) %>%
+
+      # Set fertilizer coefficient to zero when missing. This will lead to zero fertilizer cost.
+      replace_na(list(coefficient = 0)) %>%
+
+      # Calculate fertilizer cost using a fixed value (specified in constants.R in 2007$ per ton of NH3)
+      # and the fertilizer coefficient calculated above. Subtract from original nonLandVariableCost.
+      mutate(FertCost = coefficient * aglu.FERT_COST * gdp_deflator(1975, 2007) * CONV_KG_T / CONV_NH3_N ) %>%
+      mutate(nonLandVariableCost = round(nonLandVariableCost - FertCost, aglu.DIGITS_CALPRICE)) %>%
+      select(-minicam.energy.input, -coefficient, -FertCost) ->
+      L2062.AgCost_bio_irr_mgmt_adj
 
     # Produce outputs
-    # Temporary code below sends back empty data frames marked "don't test"
-    # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    L2062.AgCoef_Fert_ag_irr_mgmt %>%
+      add_title("Fertilizer coefficients for agricultural technologies") %>%
+      add_units("kgN per kg crop") %>%
+      add_comments("Map fertilizer coefficients in L142.ag_Fert_IO_R_C_Y_GLU to all technologies") %>%
+      add_comments("Note: we are using the same coefficient for all four management technologies (irrigated, rainfed, hi and lo") %>%
       add_legacy_name("L2062.AgCoef_Fert_ag_irr_mgmt") %>%
-      add_precursors("precursor1", "precursor2", "etc") %>%
-      # typical flags, but there are others--see `constants.R`
-      add_flags(FLAG_LONG_YEAR_FORM, FLAG_NO_XYEAR) ->
+      add_precursors("common/GCAM_region_names", "water/basin_to_country_mapping", "temp-data-inject/L142.ag_Fert_IO_R_C_Y_GLU") ->
```

Comment:
This is so minor because the code is very clear and Ben's comments cover a lot of what I saw. But do we want to do each precursor on its own line?